### PR TITLE
feat: empty column filter forbids all collections

### LINKED ISSUE #860: Empty column filter should forbid all collections
Today, a collection filter that is null or empty ("" string) allows all collections.
Now we want that:
- if null (no column-filter provided), then all collections are allowed
- if empty (column-filter is provided but is empty string), then all collections are forbidden
- else use its value

### PR DESCRIPTION
**Pull request by alainbodiguel**
fix #860

### CODE REVIEW

File: arlas-tests/src/test/java/io/arlas/server/tests/rest/explore/AbstractProjectedTest.java [hunk 1]
```diff
@@ -197,20 +203,34 @@ public void testReturnedGeometriesFilter() throws Exception {
...
 
         search.returned_geometries = null;
     }
 
+    @Test
+    public void testReturnedGeometriesFilterWithEmptyColumnFilter() throws Exception {
+
+        search.returned_geometries = "geo_params.geometry,geo_params.second_geometry";
+
+        handleReturnedMultiGeometries(post(search), search.returned_geometries);
```
Comment (MohamedHamouGisaia) [inline arlas-tests/src/test/java/io/arlas/server/tests/rest/explore/AbstractProjectedTest.java:224]: shouldn't this test be also a `handleUnavailableCollection` on a post query ?

File: arlas-tests/src/test/java/io/arlas/server/tests/ogc/wfs/WFSServiceIT.java [hunk 2]
```diff
@@ -265,7 +264,7 @@ public void testDescribeFeatureWithAvailableColumn() throws Exception {
                         new ImmutablePair<>("VERSION", "2.0.0"),
```
Comment (MohamedHamouGisaia) [inline arlas-tests/src/test/java/io/arlas/server/tests/ogc/wfs/WFSServiceIT.java:264]: Shouldn't we add a `handleUnavailableCollection` test as it is the case for the exploration endpoints tests ?
same for stac, raw and opensearch endpoints ?

File: arlas-tests/src/test/java/io/arlas/server/tests/ogc/csw/CSWServiceIT.java [hunk 3]
```diff
@@ -115,7 +115,7 @@ public void testGetRecordByIdWithCollectionBasedColumnFilter() throws Exception
         String id = getRecordId(collectionName);
 
         handleGetRecordById(
-                get("GetRecordById", Arrays.asList(new ImmutablePair<>("ID", id)), Optional.of(" ")),
+                get("GetRecordById", Arrays.asList(new ImmutablePair<>("ID", id)), Optional.empty()),
                 collectionName);
         handleGetRecordById(
                 get("GetRecordById", Arrays.asList(new ImmutablePair<>("ID", id)), Optional.of("panda:anyField")),
```
Comment (MohamedHamouGisaia) [inline arlas-tests/src/test/java/io/arlas/server/tests/ogc/csw/CSWServiceIT.java:121]: Shouldn't we add a `handleUnavailableCollection` test as it is the case for the exploration endpoints tests ?

**Review by mbarbet — APPROVED** [review]

File: arlas-tests/src/test/java/io/arlas/server/tests/ogc/csw/CSWServiceIT.java (hunk 3 above)
Comment (alainbodiguel) [reply arlas-tests/src/test/java/io/arlas/server/tests/ogc/csw/CSWServiceIT.java:121]: Ok

File: arlas-tests/src/test/java/io/arlas/server/tests/ogc/wfs/WFSServiceIT.java (hunk 2 above)
Comment (alainbodiguel) [reply arlas-tests/src/test/java/io/arlas/server/tests/ogc/wfs/WFSServiceIT.java:264]: ok

File: arlas-tests/src/test/java/io/arlas/server/tests/rest/explore/AbstractProjectedTest.java (hunk 1 above)
Comment (alainbodiguel) [reply arlas-tests/src/test/java/io/arlas/server/tests/rest/explore/AbstractProjectedTest.java:224]: ok

**Review by MohamedHamouGisaia — APPROVED** [review]

**Pull request merged**